integrations/lib/watcherjob/watcherjob.go

```diff
@@ -81,12 +83,21 @@ func NewJobWithEvents(events types.Events, config Config, fn EventFunc) lib.Serv
 			return nil
 		})
 
+		bk := backoff.NewDecorr(20*time.Millisecond, 5*time.Second, clockwork.NewRealClock())
+
 		log := logger.Get(ctx)
 		for {
 			err := job.watchEvents(ctx)
+			// We are not supporting liveness/readiness yet, but if we do it would make sense to use job's readiness
+			job.SetReady(false)
+
 			switch {
 			case trace.IsConnectionProblem(err):
-				log.WithError(err).Error("Failed to connect to Teleport Auth server. Reconnecting...")
+				// Not all connection problems can be retried. The client can
+				// end up in a broken state and won't be able to connect.
+				// Exiting in error is noisier but allows the orchestrator to
+				// know something is not right.
+				return trace.WrapWithMessage(err, "Failed to connect to Teleport server. Exiting.")
```

Comment:
The backoff.Do won't recover, it will only throw errors explicitly every X seconds. To recover from such broken state we have to teardown the whole client. I don't think we can do it without leaking memory and goroutines all around the place.

What we could do to mitigate is assign a retry quota, like allow 3 reconnection attempts in 5 minutes, and crash if we exceed it. But this approach is worse for users using an orchestrator as the plugin won't fail fast.

Another approach would be to configure a TELEPORT_PLUGIN_FAIL_FAST environment variable in the chart and check if the var is set in the plugin. This would be backward compatible for non-chart users.

Reply:
> Another approach would be to configure a TELEPORT_PLUGIN_FAIL_FAST environment variable in the chart and check if the var is set in the plugin. This would be backward compatible for non-chart users.

Seems like a good idea :+1:

Reply:
I implemented the feature flag in f56a732fb8b42220d771580680f15f9f7112e9d2